src/Agent.Listener/Agent.cs

```diff
@@ -132,8 +151,8 @@ public override void Initialize(IHostContext hostContext)
                     {
                         try
                         {
-                            HostContext.CancellationToken.ThrowIfCancellationRequested();
-                            message = await listener.GetNextMessageAsync(); //get next message
+                            token.ThrowIfCancellationRequested();
```

Comment:
move this to the While() loop condition. While(!token.IsCancellationRequested){}